databricks/koalas/frame.py

```diff
@@ -4371,7 +4371,12 @@ def drop(self, labels=None, axis=1,
                               for column, idx
                               in zip(self._internal.data_columns, self._internal.column_index)
                               if idx not in drop_column_index))
-            internal = self._internal.copy(data_columns=list(cols), column_index=list(idx))
+            # make column string list to drop internal spark dataframe fields
+            columns = [col[0] for col in columns]
+            internal = self._internal.copy(
+                sdf=self._sdf.drop(*columns),
```

Comment:
Actually, this seems not covering multi-index case.

I think you can just select existing data_columns:

```python
self._sdf.select(self._internal.index_scols + [self._internal.scol_for(col) for col in cols])
```

Reply:
Thanks for comment! 😃 

I just tested it for multi-index case like below,

```
>>> import databricks.koalas as ks
>>> import pandas as pd
>>> import numpy as np
>>>
>>> arrays = [[1, 1, 2, 2], ['red', 'blue', 'red', 'blue']]
>>> idx = pd.MultiIndex.from_arrays(arrays, names=('Index', 'color'))
>>> pdf = pd.DataFrame(np.random.randn(4, 5), idx)
>>> kdf = ks.from_pandas(pdf)
>>>
>>> kdf
                    0         1         2         3         4
Index color
1     red    0.304819  1.373173 -0.095708 -0.165494 -0.922387
      blue  -0.538924  0.623598  0.705721 -0.006320  1.173270
2     red    1.397902 -1.870591 -0.294745 -0.100288  0.802501
      blue   1.922724 -0.314832  1.279700  0.414461 -0.010711
>>>
>>> kdf = kdf.drop(['0', '1', '2'])
>>> kdf
                    3         4
Index color
1     red   -0.165494 -0.922387
      blue  -0.006320  1.173270
2     red   -0.100288  0.802501
      blue   0.414461 -0.010711
>>>
>>> kdf._sdf.show()
+-----+-----+--------------------+--------------------+
|Index|color|                   3|                   4|
+-----+-----+--------------------+--------------------+
|    1|  red| -0.1654939967509364|  -0.922387275283718|
|    1| blue|-0.00632027805463...|  1.1732703308776347|
|    2|  red|-0.10028785194871741|  0.8025013695853971|
|    2| blue| 0.41446113311041816|-0.01071114630144753|
+-----+-----+--------------------+--------------------+
```

I think it also works for multi-index case,

But maybe is there a something wrong that i had tested about multi-index?

Reply:
I think @HyukjinKwon meant multi-index **columns**.

Reply:
@harupy , 
You're the best! it is really helpful for me. Thanks ! 😃

Reply:
For example, `kdf.drop('e')` should remove `('e', 'a')` and `('e', 'b')`, but the current implementation does `self._sdf.drop('e')`.

Reply:
@harupy ,Really appreciate for your helping. Now maybe i really finished. could you check this out if when you available??

Reply:
@harupy Could i use your multi-index example as an additional example for `DataFrame.drop()` maybe if you don't mind??

Reply:
@itholic 
Yes of courser you can 😃😃😃

Reply:
@harupy Thanks. i just added it. 😸